analyzer/evmtokenbalances/evm_token_balances.go

```diff
@@ -78,25 +80,28 @@ const (
 )
 
 type Main struct {
-	runtime common.Runtime
-	source  storage.RuntimeSourceStorage
-	target  storage.TargetStorage
-	logger  *log.Logger
+	runtime         common.Runtime
+	runtimeMetadata sdkConfig.ParaTime
+	source          storage.RuntimeSourceStorage
+	target          storage.TargetStorage
+	logger          *log.Logger
 }
 
 var _ analyzer.Analyzer = (*Main)(nil)
 
 func NewMain(
 	runtime common.Runtime,
+	runtimeMetadata sdkConfig.ParaTime,
```

Comment:
most of the codebase uses *sdkConfig.ParaTime. do we specifically want a copy here? or if copying is always better, should we change many things over? there are lots of places where there's a structure that's not especially large or mutable and where we neither want separate copies nor a shared copy.

Reply:
Changed to pointers for consistency. I tend to prefer copies over pointers, because pointers imply mutability or nullability, or (unfortunately) the need to hand-hold the compiler and prevent expensive copies. I'm hoping that the compiler can get it right most of the time and avoid an actual copy, but I don't know how true that is.